lib/cloud-operation/cloud-operation-factory.ts

```diff
@@ -0,0 +1,19 @@
+export class CloudOperationFactory implements ICloudOperationFactory {
```

Comment:
[Suggestion] I don't really see the point in ICloudOperationFactory, ICloudCommunicationChannelFactory and IWebSocketFactory - I think you can use IFactory for all three